src/TypeParser/JMSTypeParser.php

```diff
@@ -91,4 +94,13 @@ private function parseType(array $typeInfo, bool $isSubType = false): PropertyTy
 
         throw new InvalidTypeException(sprintf('Unknown JMS property found (%s)', var_export($typeInfo, true)));
     }
+
+    private function isCollection(string $name): bool
+    {
+        if (self::TYPE_ARRAY_COLLECTION === $name || Collection::class === $name) {
+            return true;
+        }
+
+        return is_subclass_of($name, Collection::class);
```

Comment:
```suggestion
        return is_a($name, Collection::class, true);
```

Reply:
i'd merge the 2 conditions and do `return self::TYPE_ARRAY_COLLECTION === $name || is_a($name, Collection:class, true);`

Reply:
Fixed via 74742d61d256a41dea7dfd43fade461a478bfbf0